contracts/cw-law-stone/examples/multiple-sources/template.pl

```diff
@@ -0,0 +1,32 @@
+valid_did(DID, Addr) :-
+    did_components(DID, did(Method, Addr, _, _, _)),
+    allow_did_method(Method),
+    allow_addr(Addr).
+
+min_amount(Action, MinAmount) :-
+    Action == exec_workflow,
+    min_exec_workflow_amount(MinAmount).
+
+min_amount(Action, MinAmount) :-
+    Action == create_dataset,
+    min_create_dataset_amount(MinAmount).
+
+min_amount(Action, MinAmount) :-
+    Action == create_service,
+    min_create_service_amount(MinAmount).
+
+has_sufficient_balance(Addr, MinAmount) :-
+    bank_spendable_balances(Addr, Balances),
+    member(Denom-Amount, Balances),
+    allow_denom(Denom),
+    Amount @>= MinAmount.
+
+can(Action, DID) :-
+    Action == change_governance,
+    valid_did(DID, Addr),
+    admin_addr(Addr).
```

Comment:
You can put the term in the head clause instead of doing comparison. It will make the code easier to read (this is a general principle).

```suggestion
can(change_governance, DID) :-
    valid_did(DID, Addr),
    admin_addr(Addr).
```